### ISSUE TITLE
Addition of Actor-Critic model for Reinforcement Learning.

### ISSUE THREAD

**Issue by naruarjun** [mlpack/mlpack#2262]
I have started working on the implementation of a actor-critic model just like the Q learning model in the reinforcement learning methods. It'll basically be a addition of algorithms mentioned in this blog post. 
https://towardsdatascience.com/understanding-actor-critic-methods-931b97b6df3f.
I have gone through the relevant papers and have already implemented this before using tensorflow (https://github.com/naruarjun/Reinforcement_Learning).

I wanted to ask if this is a worthwhile addition to the library and if so I will continue to implement it and finish ASAP.

**Comment by birm** [comment]
Are you looking to add missing layers/functions/components to make this model possible, or the model itself? If the former, we'd welcome a PR on this repository; if the later, then see https://github.com/mlpack/models

**Comment by naruarjun** [comment]
Yes I am referring to the former. It'll be code to create a agent similar to the q learning agent code already present in mlpack.

**Comment by codeboy5** [comment]
Hi, @naruarjun are you working on this issue?

**Comment by naruarjun** [comment]
Yes I am, I’ll make a PR within this week.

**Comment by naruarjun** [comment]
I have also made a roadmap of how the implementation is done. I’ll post that here once I get home to indicate the progress I’ve made.

**Comment by naruarjun** [comment]
Roadmap:-
1) The main functions involved will be the same as the q learning agent i.e. Step,Episode.
2) A custom replay class will be made that can replay one whole episode at a time.
3) Training will occur in the Episode function rather than the step function.
4) Training is similar to the episode function here(https://github.com/naruarjun/Reinforcement_Learning/blob/master/Cartpole/policygradient.py)
5)So the agent will step through one episode with the data for that episode stored, train via policy gradients with advantage.

Any suggestions for improvements are welcome. The PR will be coming in a few days.
